### PR TITLE
[Peter Latona] - Deploy and Interact with Smart Contracts on Celo Blockchain

### PR DESCRIPTION
**Pull request by SasaniEldis**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+Step 1: Open your text editor and create a new file.
+
+Step 2: Write the code for your smart contract in Solidity.
+
+For example, you could write a simple smart contract that stores a string:
+
+**C#>>**
+
+```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:77]: > You Made double time ``` Also If its a solidity code start with ```solidity

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+3. Compiling the Smart Contract
+
+After creating file extensions for operation purposes e.g saving strings, compile the smart contract by following these steps:
+
+Step 1: Open your terminal and navigate to the smart contract file's directory.
+
+Step 2: Run the following command to compile the smart contract:
+
+**CSS>>**
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:100]: Is CSS write Or There is something else to be here?

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+You can write a smart contract by following these steps:
+
+Step 1: Open your text editor and create a new file.
+
+Step 2: Write the code for your smart contract in Solidity.
+
+For example, you could write a simple smart contract that stores a string:
+
+**C#>>**
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:75]: What code is in C#?

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+Step 1: Start the Ganache CLI by running the following command in your terminal:
+
+ganache-cli
+
+Step 2: Open a new terminal window and navigate to the smart contract file's directory.
+
+Step 3: Run the following command to deploy the smart contract:
+
+**Python>>**
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:129]: if I am not wrong you adding this for showing what kind of code it is? you can just use ```bash if its a command ```solidity if its a solidity code ```js for javascript

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+Step 1: Open a new file in your text editor.
+
+Step 2: Write the code to interact with the smart contract.
+
+For example, you could write a script that sets the value of the string in the smart contract:
+
+**Javascript>>**
+
+```
+ ```const Web3 = require('web3');
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:155]: `You Also Did double time ``` here `

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+});```
+```
+
+Replace &lt;YOUR-CELO-RPC-ENDPOINT>, &lt;YOUR-CONTRACT-ADDRESS>, and &lt;YOUR-CELO-ADDRESS> with your Celo RPC endpoint, contract address, and Celo address, respectively.
+
+Step 3: Save the file with a .js extension (e.g., interact.js).
+
+Step 4: Run the script by running the following command in your terminal:
+
+```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:180]: ` Make it ```bash `

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md (hunk 4 above)
Comment (SasaniEldis) [reply blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:129]: Alright. Will make the edits where necessary

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+You can write a smart contract by following these steps:
+
+Step 1: Open your text editor and create a new file.
+
+Step 2: Write the code for your smart contract in Solidity.
+
+For example, you could write a simple smart contract that stores a string:
+
+Solidity:
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:75]: It should not be there

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+  ```pragma solidity ^0.8.0;
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:77]: ` it should be ```solidity
Then next line 
pragma solidity ^0.8.0 `

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+
+contract MyContract {
+    string public myString;
+
+    function setMyString(string memory _myString) public {
+        myString = _myString;
+    }
+}```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:85]: ` it should be }
Then next line ``` 
`

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+3. Compiling the Smart Contract
+
+After creating file extensions for operation purposes e.g saving strings, compile the smart contract by following these steps:
+
+Step 1: Open your terminal and navigate to the smart contract file's directory.
+
+Step 2: Run the following command to compile the smart contract:
+
+ 
+Bash:
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:100]: Bash not here

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+
+```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:102]: ` it should be like ```bash
`

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+
+Step 1: Start the Ganache CLI by running the following command in your terminal:
+
+ganache-cli
+
+Step 2: Open a new terminal window and navigate to the smart contract file's directory.
+
+Step 3: Run the following command to deploy the smart contract:
+
+Bash:
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:127]: As i mentioned upside for bash same goes for all bash code

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+
+Step 1: Open a new file in your text editor.
+
+Step 2: Write the code to interact with the smart contract.
+
+For example, you could write a script that sets the value of the string in the smart contract:
+
+Js:
+
+ ```const Web3 = require('web3');
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:152]: ` it should be ```js then next line 
const Web3 = require('web3');

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+    '<YOUR-CONTRACT-ADDRESS>'
+);
+
+myContract.methods.setMyString('Hello, World!').send({
+    from: '<YOUR-CELO-ADDRESS>',
+    gas: 500000
+})
+.then((receipt) => {
+    console.log(receipt);
+});```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:168]: ` it should be }); then next line  ```
`

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:169]: This one is extra

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+## Common Errors Encountered and How to Fix them
+
+Here are some common errors you might encounter while following this tutorial, along with their corrections:
+
+
+1. Error: "Cannot find module 'web3'"
+
+This error occurs when the web3 module is not installed in your project. To fix it, run the following command in your terminal:
+
+```
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:192]: `Make it  ```bash`

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+    title: contributor, Celo sages
+url: https://github.com/SasaniEldis
+image_url: https://avatars.githubusercontent.com/u/107583957?v=4
+tags: ["beginner", "smart contracts", "celosage", "blockchain", "solidity"]
+hide_table_of_contents: true
+slug: /tutorials/deploy-and-interact-with-smart-contracts-on-the-celo-blockchain
+---
+![header](../../src/data-tutorials/showcase/beginner/deploy-and-Interact-with-Smart-Contracts-on-the-Celo-Blockchain.png)
+
+# Deploy and Interact with Smart Contracts on the Celo Blockchain
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:15]: `Add one more # in all headline`

File: blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+
+This tutorial taught us how to deploy and interact with smart contracts on the Celo blockchain. We started by setting up a local development environment, and proceeded to write a simple smart contract in Solidity, compiled and deployed the smart contract using the Celo CLI, and finally interacted with the smart contract using a Node.js script.
+
+# Next Steps:
+
+Now that you have deployed and interacted with a simple smart contract on the Celo blockchain, you can explore more advanced features and functionalities of smart contracts, such as; event logging, function modifiers, and inheritance. You can also explore other tools and frameworks for building decentralized applications on the Celo network, such as Truffle and Hardhat. Enjoy coding!
+
+# About the Author:
+
+# Imole Peter L.
```
Comment (developerkunal) [inline blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md:236]: Name should not be a header.

File: src/data-tutorials/users.tsx [hunk 19]
```diff
@@ -938,6 +938,14 @@ const Users: User[] = [
     website: '/blog/tutorials/dynamic-nft-creation-using-svg-to-build-in-smart-contract-on-celo',
     tags: ['intermediate', 'solidity','celosage','erc721','truffle'],
   },
+  {
+    title: 'Deploy and Interact with Smart Contracts on the Celo Blockchain',
```
Comment (developerkunal) [inline src/data-tutorials/users.tsx:942]: You already added this block down but now added this now 2 times

Comment (SasaniEldis) [reply src/data-tutorials/users.tsx]: @developerkunal 

I removed one already. Thanks

**Comment by developerkunal** [comment]
@SasaniEldis Just Fix the Headers It should be ## for headings and ### for subheading 
and second thing use the https://prettier.io/playground before pushing.

**Comment by viral-sangani** [comment]
Closing Celo Sage PRs

**Pull request closed without merging**